core/frontend/src/types/autopilot/parameter-table.ts

```diff
@@ -0,0 +1,110 @@
+import { isNumber } from 'lodash'
+
+import * as arducopter_metadata from '@/assets/parameter-metadata/arducopter-4.2.pdef.json'
+import * as ardurover_metadata from '@/assets/parameter-metadata/ardurover-4.2.pdef.json'
+import * as ardusub_metadata from '@/assets/parameter-metadata/ardusub-4.1.pdef.json'
+import Notifier from '@/libs/notifier'
+import autopilot from '@/store/autopilot_manager'
+import { Dictionary } from '@/types/common'
+
+import { parameters_service } from '../frontend_services'
+import Parameter from './parameter'
+
+const notifier = new Notifier(parameters_service)
+
+// Parameter metadata as in the JSON files
+interface Metadata {
+  Description?: string
+  DisplayName: string
+  Increment?: string
+  Range?: {
+    high: string
+    low: string
+  }
+  RebootRequired?: string,
+  ReadOnly?: string,
+  Bitmask?: {[key:number] : string}
+  Values?: {[key:number] : string}
+  User?: string
+}
+
+interface MetadataCategory {
+  [key: string]: Metadata | number; // number deals with a sneaky {"json": 0} entry
+}
+interface MetadataFile {
+    [key: string]: MetadataCategory;
+}
+export default class ParametersTable {
+    parametersDict: {[key: number] : Parameter} = {}
+
+    count = 0
+
+    metadata = {} as Dictionary<Metadata>
+
+    constructor() {
+      this.fetchMetadata()
+    }
+
+    fetchMetadata(): void {
+      if (autopilot.vehicle_type === null) {
+        // Check again later if we have a vehicle type identified
+        setTimeout(() => { this.fetchMetadata() }, 1000)
+        return
+      }
+      // default to submarine
+      let metadata = ardusub_metadata as MetadataFile
+      // This is to avoid importing a 40 lines enum from mavlink and adding a switch case with 40 cases
+      if (autopilot.vehicle_type.toLowerCase().includes('copter')
+      || autopilot.vehicle_type.toLowerCase().includes('rotor')) {
+        metadata = arducopter_metadata
+      } else if (autopilot.vehicle_type.toLowerCase().includes('rover')
+      || autopilot.vehicle_type.toLowerCase().includes('boat')) {
+        metadata = ardurover_metadata
+      }
```

Comment:
From our meeting discussion -> this will likely get handled by querying the known firmware type (can't remember if that was in an existing `store` or something).

Reply:
yes, I think I'll do this in a separate pr